# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ffmpeg/x264-core65.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ffmpeg/libavcodec/libx264.c.vanilla	2008-11-18 11:55:07.000000000 +0100
+++ ffmpeg/libavcodec/libx264.c	2008-11-18 11:57:35.000000000 +0100
@@ -160,4 +160,8 @@
     x4->params.i_bframe = avctx->max_b_frames;
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
+#if X264_BUILD > 62
+    x4->params.i_bframe_adaptive = avctx->b_frame_strategy;
+#else
     x4->params.b_bframe_adaptive = avctx->b_frame_strategy;
+#endif
     x4->params.i_bframe_bias = avctx->bframebias;
@@ -227,8 +232,11 @@
     x4->params.analyse.i_me_range = avctx->me_range;
     x4->params.analyse.i_subpel_refine = avctx->me_subpel_quality;
 
+#if X264_BUILD < 65
     x4->params.analyse.b_bidir_me = avctx->bidir_refine > 0;
     x4->params.analyse.b_bframe_rdo = avctx->flags2 & CODEC_FLAG2_BRDO;
+#endif
+
     x4->params.analyse.b_mixed_references =
         avctx->flags2 & CODEC_FLAG2_MIXED_REFS;
     x4->params.analyse.b_chroma_me = avctx->me_cmp & FF_CMP_CHROMA;
@@ -261,7 +260,9 @@
     x4->params.rc.f_ip_factor = 1/fabs(avctx->i_quant_factor);
     x4->params.rc.f_pb_factor = avctx->b_quant_factor;
     x4->params.analyse.i_chroma_qp_offset = avctx->chromaoffset;
+#if X264_BUILD < 62
     x4->params.rc.psz_rc_eq = avctx->rc_eq;
+#endif
 
     x4->params.analyse.b_psnr = avctx->flags & CODEC_FLAG_PSNR;
     x4->params.i_log_level = X264_LOG_DEBUG;
